txpipe/twopoint.py

```diff
@@ -705,297 +554,9 @@ def read_metadata(self):
-        for b1, b2 in s.get_tracer_combinations(cl_ee):
-            source_tracers.add(b1)
-            source_tracers.add(b2)
-
-        lens_tracers = set()
-        for b1, b2 in s.get_tracer_combinations(cl_density):
-            lens_tracers.add(b1)
-            lens_tracers.add(b2)
-
-
-        return len(source_tracers), len(lens_tracers)
-
-
-    def replace_with_theory_fourier(self, s, cosmo):
-
-        import pyccl
-
-        nbin_source, nbin_lens = self.read_nbin(s)
-        tracers = self.get_ccl_tracers(s, cosmo)
-
-        data_types = s.get_data_types()
-        if 'galaxy_shearDensity_cl_b' in data_types:
-            # Remove galaxy_shearDensity_cl_b measurement values
-            ind_b = s.indices('galaxy_shearDensity_cl_b')
-            s.remove_indices(ind_b)
-        if 'galaxy_shear_cl_bb' in data_types:
-            # Remove galaxy_shear_cl_bb  measurement values
-            ind_bb = s.indices('galaxy_shear_cl_bb')
-            s.remove_indices(ind_bb)
-
-        for i in range(nbin_source):
-            for j in range(i+1):
-                print(f"Computing theory lensing-lensing ({i},{j})")
-
-                # compute theory
-                print(tracers[f'source_{i}'], tracers[f'source_{j}'])
-                ell, *_  = s.get_ell_cl('galaxy_shear_cl_ee', f'source_{i}' , f'source_{j}')
-                cl = pyccl.angular_cl(cosmo, tracers[f'source_{i}'], tracers[f'source_{j}'], ell)
-
-                # replace data values in the sacc object for the theory ones
-                ind = s.indices('galaxy_shear_cl_ee', (f'source_{i}', f'source_{j}'))
-                for p, q in enumerate(ind):
-                    s.data[q].value = cl[p]
-
-
-        for i in range(nbin_lens):
-            print(f"Computing theory density-density ({i},{i})")
-
-            # compute theory
-            ell, *_  = s.get_ell_cl('galaxy_density_cl', f'lens_{i}' , f'lens_{i}')
-            cl = pyccl.angular_cl(cosmo, tracers[f'lens_{i}'], tracers[f'lens_{i}'], ell)
-
-            # replace data values in the sacc object for the theory ones
-            ind = s.indices('galaxy_density_cl', (f'lens_{i}', f'lens_{i}'))
-            for p, q in enumerate(ind):
-                s.data[q].value = cl[p]
-
-        for i in range(nbin_source):
-
-            for j in range(nbin_lens):
-                print(f"Computing theory lensing-density (S{i},L{j})")
-
-                # compute theory
-                ell, *_ = s.get_ell_cl('galaxy_shearDensity_cl_e', f'source_{i}' , f'lens_{j}')
-                cl = pyccl.angular_cl(cosmo, tracers[f'source_{i}'], tracers[f'lens_{j}'], ell)
-
-                # replace data values in the sacc object for the theory ones
-                ind = s.indices('galaxy_shearDensity_cl_e', (f'source_{i}', f'lens_{j}'))
-                for p, q in enumerate(ind):
-                    s.data[q].value = cl[p]
-
-        return s
-
-
-
+    
 
+    
 class TXTwoPointPlots(PipelineStage):
     """
     Make n(z) plots
```

Comment:
this is to make the DV plots right? also wondering if this should be split off to its own function now that theory is split off?

Reply:
It is - yep - I've fixed the docstring.  Not quite sure which part you mean should be split off?